en/vespa9-release-notes.html

```diff
@@ -0,0 +1,350 @@
+  </tr>
+  </thead>
+  <tbody>
+  </tbody>
+</table>
+
+
+<h3 id="changed-semantics-in-services-xml">Changed semantics in services.xml</h3>
+<p>
+  The following elements and attributes in services.xml have new semantics:
+</p>
+<table class="table">
+  <thead>
+  <tr>
+    <th>Name</th><th>Description</th>
+  </tr>
+  </thead>
+  <tbody>
+  </tbody>
+</table>
+
+
+<h3 id="removed-constructs-from-services-xml">Removed constructs from services.xml</h3>
+<p>
+  The following elements and attributes are removed from services.xml:
+</p>
+<table class="table">
+  <thead>
+  <tr>
+    <th>Parent element</th><th>Removed construct</th><th>Description</th>
+  </tr>
+  </thead>
+  <tbody>
+  </tbody>
+</table>
+
+
+
+<h3 id="searchdefinitions-folder"><em>searchdefinitions/</em> folder support removed</h3>
+<p>
+  Schemas should now be placed in the <em>schemas/</em> folder.
+</p>
+
+
+<h2 id="java-api-changes">Java API changes</h2>
+
+<h3 id="removed-java-packages">Removed Java packages</h3>
+
+<table class="table">
+  <thead><tr><th>Package</th><th>Description</th></tr></thead>
+  <tbody>
+  </tbody>
+</table>
+
+
+<h3 id="removed-java-classes-and-methods">Removed Java Classes and methods</h3>
+
+<p>
+  Classes and methods that were marked as deprecated in Vespa 8 are removed.
+  If deprecation warnings are emitted for Vespa APIs when building the application,
+  these must be fixed before migrating to Vespa 9. The sections below contain only the
+  most notable changes.
+</p>
+
+<p>
+  The following classes are no longer public API and have been moved to Vespa internal packages:
+</p>
+<table class="table">
+  <thead><tr><th>Package</th><th>Class</th><th>Migration advice</th></tr></thead>
+  <tbody>
+</table>
+
+<p>
+  The following methods are removed:
+</p>
+<table class="table">
+  <thead><tr><th>Method</th><th>Migration advice</th></tr></thead>
+  <tbody>
+  </tbody>
+</table>
+
+
+<h3 id="changed-java-apis">Breaking changes to Java APIs</h3>
+<p>
+  The Javadoc of the deprecated types/members should document the replacement API.
+  The below list is not exhaustive - some smaller and trivial changes are not listed.
+</p>
+<table class="table">
+  <thead><tr><th>Type(s)</th><th>Description</th></tr></thead>
+  <tbody>
+  </tbody>
+</table>
+
+
+<h3 id="deprecated-java-apis">Deprecated Java APIs</h3>
+<p>
+  A few redundant APIs have been deprecated because they have replacements that
+  provide the same, or better, functionality. We advise you switch to the
+  replacement to reduce future maintenance cost.
+</p>
+
+<table class="table">
+  <thead><tr><th>Type(s)</th><th>Replacement</th></tr></thead>
+  <tbody>
+  </tbody>
+</table>
+
+
+
+
+<h2 id="container-runtime">Container Runtime Environment</h2>
+
+
+<h3 id="jdk-version">JDK version</h3>
+<p>
+  Vespa 9 upgrades the JDK version from 17 to 25 (?). To ensure full compatibility, all container components
+  should be rebuilt with JDK 25 before being deployed on Vespa 9.
```

Comment:
Java artifacts built against older JDK versions will still be compatible with Vespa 9 (JDK 25). The opposite will not be possible - Vespa 8 (JDK 21) is not compatible with newer JVM byte code. It's possible though to use the   `--release` option for `javac` to target an older JDK version.